### DISCUSSION TITLE
Stuck at "Failed to build wheel for memray"

### DISCUSSION THREAD

**Discussion by mandyq1000** [bloomberg/memray#330] [Q&A] [answered]
#15 112.1 Failed to build memray
#15 112.1 ERROR: Could not build wheels for memray, which is required to install pyproject.toml-based projects

Any help would be much appreciated 
I am trying to install it on a docker image with a base of python 3.9
------- 
adding more error logs if anyone can help me solve it 
 libtool: link: ar cru .libs/libbacktrace.a  atomic.o dwarf.o fileline.o posix.o print.o sort.o state.o backtrace.o simple.o elf.o mmapio.o mmap.o
#15 110.8       ar: `u' modifier ignored since `D' is the default (see `U')
#15 110.8       libtool: link: ranlib .libs/libbacktrace.a
#15 110.8       libtool: link: ( cd ".libs" && rm -f "libbacktrace.la" && ln -s "../libbacktrace.la" "libbacktrace.la" )
#15 110.8       make[1]: Leaving directory '/tmp/tmpwb6uzy4o'
#15 110.8       executing command: `make install`
#15 110.8       make[1]: Entering directory '/tmp/tmpwb6uzy4o'
#15 110.8        /bin/mkdir -p '/tmp/pip-install-mb9csqy0/memray_2e99955dbeaa4e529a3934daa0f4140a/src/vendor/libbacktrace/install/lib'
#15 110.8        /bin/bash ./libtool   --mode=install /usr/bin/install -c   libbacktrace.la '/tmp/pip-install-mb9csqy0/memray_2e99955dbeaa4e529a3934daa0f4140a/src/vendor/libbacktrace/install/lib'
#15 110.8       libtool: install: /usr/bin/install -c .libs/libbacktrace.lai /tmp/pip-install-mb9csqy0/memray_2e99955dbeaa4e529a3934daa0f4140a/src/vendor/libbacktrace/install/lib/libbacktrace.la
#15 110.8       libtool: install: /usr/bin/install -c .libs/libbacktrace.a /tmp/pip-install-mb9csqy0/memray_2e99955dbeaa4e529a3934daa0f4140a/src/vendor/libbacktrace/install/lib/libbacktrace.a
#15 110.8       libtool: install: chmod 644 /tmp/pip-install-mb9csqy0/memray_2e99955dbeaa4e529a3934daa0f4140a/src/vendor/libbacktrace/install/lib/libbacktrace.a
#15 110.8       libtool: install: ranlib /tmp/pip-install-mb9csqy0/memray_2e99955dbeaa4e529a3934daa0f4140a/src/vendor/libbacktrace/install/lib/libbacktrace.a
#15 110.8       libtool: finish: PATH="/tmp/pip-build-env-v53j7qa7/overlay/bin:/tmp/pip-build-env-v53j7qa7/normal/bin:/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/sbin" ldconfig -n /tmp/pip-install-mb9csqy0/memray_2e99955dbeaa4e529a3934daa0f4140a/src/vendor/libbacktrace/install/lib
#15 110.8       ----------------------------------------------------------------------
#15 110.8       Libraries have been installed in:
#15 110.8          /tmp/pip-install-mb9csqy0/memray_2e99955dbeaa4e529a3934daa0f4140a/src/vendor/libbacktrace/install/lib
#15 110.8       
#15 110.8       If you ever happen to want to link against installed libraries
#15 110.8       in a given directory, LIBDIR, you must either use libtool, and
#15 110.8       specify the full pathname of the library, or use the `-LLIBDIR'
#15 110.8       flag during linking and do at least one of the following:
#15 110.8          - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
#15 110.8            during execution
#15 110.8          - add LIBDIR to the `LD_RUN_PATH' environment variable
#15 110.8            during linking
#15 110.8          - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
#15 110.8          - have your system administrator add LIBDIR to `/etc/ld.so.conf'
#15 110.8       
#15 110.8       See any operating system documentation about shared libraries for
#15 110.8       more information, such as the ld(1) and ld.so(8) manual pages.
#15 110.8       ----------------------------------------------------------------------
#15 110.8        /bin/mkdir -p '/tmp/pip-install-mb9csqy0/memray_2e99955dbeaa4e529a3934daa0f4140a/src/vendor/libbacktrace/install/include/libbacktrace'
#15 110.8        /usr/bin/install -c -m 644 /tmp/pip-install-mb9csqy0/memray_2e99955dbeaa4e529a3934daa0f4140a/src/vendor/libbacktrace/backtrace.h backtrace-supported.h /tmp/pip-install-mb9csqy0/memray_2e99955dbeaa4e529a3934daa0f4140a/src/vendor/libbacktrace/internal.h /tmp/pip-install-mb9csqy0/memray_2e99955dbeaa4e529a3934daa0f4140a/src/vendor/libbacktrace/debuginfod_support.h '/tmp/pip-install-mb9csqy0/memray_2e99955dbeaa4e529a3934daa0f4140a/src/vendor/libbacktrace/install/include/libbacktrace'
#15 110.8       make[1]: Leaving directory '/tmp/tmpwb6uzy4o'
#15 110.8       building 'memray._memray' extension
#15 110.8       creating build/temp.linux-aarch64-cpython-39
#15 110.8       creating build/temp.linux-aarch64-cpython-39/src
#15 110.8       creating build/temp.linux-aarch64-cpython-39/src/memray
#15 110.8       creating build/temp.linux-aarch64-cpython-39/src/memray/_memray
#15 110.8       gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -D_FILE_OFFSET_BITS=64 -DUSE_MEMRAY_TLS_MODEL=1 -Isrc/memray/_memray -Isrc -I/tmp/pip-install-mb9csqy0/memray_2e99955dbeaa4e529a3934daa0f4140a/src/vendor/libbacktrace/install/include -I/usr/local/include/python3.9 -c src/memray/_memray.cpp -o build/temp.linux-aarch64-cpython-39/src/memray/_memray.o -std=c++17 -Wall -flto -fvisibility=hidden
#15 110.8       In file included from src/memray/_memray/source.h:9,
#15 110.8                        from src/memray/_memray.cpp:849:
#15 110.8       src/memray/_memray/lz4_stream.h:35:10: fatal error: lz4frame.h: No such file or directory
#15 110.8          35 | #include <lz4frame.h>
#15 110.8             |          ^~~~~~~~~~~~
#15 110.8       compilation terminated.
#15 110.8       error: command '/usr/bin/gcc' failed with exit code 1
#15 110.8       [end of output]
#15 110.8   
#15 110.8   note: This error originates from a subprocess, and is likely not a problem with pip.
#15 110.8   ERROR: Failed building wheel for memray

**Comment by pablogsal** [comment, marked as answer]
Hi @mandyq1000 and thanks for checking with us.

It seems that you are trying to use `memray` in a system where there isn't a wheel available or with an old interpreter and you are building it from source. The error you are seeing is because you are missing one of the binary dependencies or your build system cannot find the headers (and likely also the libraries). Could you give us more information on the system you are using and the details of your interpreter so we can tell you how to build from source?
